modules/objc/pages/vector-search-api-reference.adoc

```diff
@@ -0,0 +1,286 @@
+}
+
+----
+
+IMPORTANT: The beta release currently only supports the format of vectors as an array of numbers.
+
+=== Default Value Constants
+
+[source, c]
+----
+
+public class VectorIndexConfiguration {
+  /* Scalar Quantizer */
+  public static let defaultEncoding = VectorEncoding.scalarQuantizer()
+
+  /* Squared Euclidean Distance */
+  public static let defaultMetric = VectorMetric.euclidean   
+}
+
+----
+
+=== Extension Path Configuration
+
+NOTE: The extension configuration is only necessary for the CBL-iOS MacOS and CBL-C platforms as dynamic extension libraries may be at arbitrary locations.
+
+[source, c]
+----
+
+public class CouchbaseLite {
+  /* Registers a directory path to load extension libraries from. Must 
+     be called before using CouchbaseLite but only when using extensions */
+  public static func setExtensionPath(extensionPath: String)
+}
+
+----
+
+== Vector SQL++ Functions
+
+Couchbase Lite currently supports two SQL++ functions, `vector_match()` and `vector_distance()`.
+
+=== Vector Match
+
+==== `vector_match(vectorIndexIdentifier, targetVectorExpr, [limit = 3])`
+
+[cols = "3*"]
+|===
+|Parameter |Is Required |Description
+
+|vectorIndexIdentifier
+|Required
+|The name of the vector index to perform the vector search on.
+|targetVectorExpr
+|Required
+|The target vector expression that returns a vector in the form of an array of numbers.
+|limit
+|Optional
+|The limit number of the returned matched results.
+The maximum number allowed is 10000. An error will be returned when creating a query with a limit greater than 10000.
+
+|===
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
+If the specified index does not exist, an error will occur on creation of the query.
+The matched vectors will be returned up to the specified limit number, if the limit is not specified then the default value will be used. 
+The returned vectors are sorted by their distance values in ascending order by default.
+
+NOTE: The default value for the `limit` parameter is 3.
+
+[source, sqlpp]
+----
+SELECT name 
+FROM photos 
+WHERE vector_match(photos-index, [0.1, 0.5, 1.2])
+
+----
+
+[source, sqlpp]
+----
+
+SELECT name 
+FROM photos
+WHERE vector_match(photos-index, [0.1, 0.5, 1.2]) AND city = "San Francisco"
+
+----
+
+=== Vector Distance
+
+==== `vector_distance(vectorIndexIdentifier)`
+
+[cols = "3*"]
+|===
+|Parameter |Is Required |Description
+
+|vectorIndexIdentifier
+|Required
+|The name of the vector index.
+
+|===
+
+This function returns the distance between the target vector specified in the `vector_match()` function and the matched vector in the specified vector index based on the distance metric set in the index configuration .
+
+[source, sqlpp]
+----
+
+SELECT name, vector_distance(photos-index) 
+FROM photos
+WHERE vector_match(photos-index, [0.1, 0.5, 1.2])
+
+----
+
+[source, sqlpp]
+----
+
```

Comment:
Shouldn't we describe these examples in context of API? Or is that covered in different section